extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java

```diff
@@ -92,6 +102,27 @@ void announceFeature(BuildProducer<FeatureBuildItem> features) {
         features.produce(new FeatureBuildItem(Feature.REST_CLIENT_REACTIVE));
     }
 
+    @BuildStep
+    void registerServiceProviders(BuildProducer<ServiceProviderBuildItem> services) {
+        services.produce(new ServiceProviderBuildItem(io.smallrye.stork.config.ConfigProvider.class.getName(),
+                MicroProfileConfigProvider.class.getName()));
+
+        for (Class<?> providerClass : asList(LoadBalancerProvider.class, ServiceDiscoveryProvider.class)) {
+            String service = "META-INF/services/" + providerClass.getName();
+
+            // find out all the implementation classes listed in the service files
+            Set<String> implementations;
+            try {
+                implementations = ServiceUtil.classNamesNamedIn(Thread.currentThread().getContextClassLoader(),
+                        service);
+            } catch (IOException e) {
+                throw new IllegalStateException("Failed to collect implementations of " + service, e);
+            }
+
+            services.produce(new ServiceProviderBuildItem(providerClass.getName(), implementations));
```

Comment:
Can't we use `ServiceProviderBuildItem#allProviders`?

Reply:
`allProviders` focuses on a single file. But I see there's also `allProvidersFromClassPath` which does exactly what I need here :)